### PR TITLE
AA-382: Inject system events

### PR DESCRIPTION
**Pull request by forshtat**

### CODE REVIEW

File: tests/rip7560/test_send_failed.py [hunk 1]
```diff
@@ -126,6 +126,16 @@ def test_eth_send_account_validation_reverts1(
     assert_rpc_error(response, case.expected_message, -32000, expected_data)
 
 
+def encode_panic_error(w3, value):
```
Comment (drortirosh) [inline tests/rip7560/test_send_failed.py:129]: this is not a "panic" Panic(uint256), but "standard error" Error(string)

File: tests/rip7560/test_rip7560.py [hunk 2]
```diff
@@ -38,6 +41,147 @@ def test_eth_sendTransaction7560_valid1(w3, wallet_contract, tx_7560):
...
+        "sender": new_sender_address,
+        "paymaster": paymaster_contract_7560.address,
+        "deployer": factory_contract_7560.address,
+        # pylint: disable=fixme
+        # TODO: also pass and check 'nonceKey'
+        "nonce": 0,
+        "success": True,
+        # pylint: disable=fixme
+        # TODO: pass and check real 'actualGasCost'
+        "actualGasCost": 0,
```
Comment (drortirosh) [inline tests/rip7560/test_rip7560.py:81]: not checking exact gas you can use `pytest.approx()`

File: tests/rip7560/test_rip7560.py [hunk 3]
```diff
@@ -38,6 +41,147 @@ def test_eth_sendTransaction7560_valid1(w3, wallet_contract, tx_7560):
     w3.eth.get_transaction(rethash)
 
 
+def test_system_event_success(
+    w3, entry_point_rip7560, factory_contract_7560, paymaster_contract_7560, tx_7560
+):
+    tx_7560.paymaster = paymaster_contract_7560.address
```
[the remark this answers is not in the document]
Comment (forshtat) [reply tests/rip7560/test_rip7560.py]: Done.

File: tests/rip7560/test_rip7560.py [hunk 4]
```diff
@@ -38,6 +41,147 @@ def test_eth_sendTransaction7560_valid1(w3, wallet_contract, tx_7560):
...
+
+    tx_7560.send()
+    send_bundle_now()
+
+    system_event_args, execution_revert_event_args, _ = get_system_events(
+        entry_point_rip7560
+    )
+
+    expected_revert_reason = bytes.fromhex(encode_panic_error(w3, "reverting")[2:])
+    assert execution_revert_event_args == {
```
[the remark this answers is not in the document]
Comment (forshtat) [reply tests/rip7560/test_rip7560.py:127]: Fixed.

File: tests/rip7560/test_send_failed.py (hunk 1 above)
Comment (forshtat) [reply tests/rip7560/test_send_failed.py]: Fixed.

File: tests/rip7560/test_rip7560.py (hunk 2 above)
Comment (forshtat) [reply tests/rip7560/test_rip7560.py]: Removed `actualGasCost` from the event.

**Review by shahafn — APPROVED** [review]

**Pull request merged**